### PR TITLE
ADEN-7698 Fix Cheshire Cat tracking and bugs

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: app/modules/ads/module.js [hunk 1]
```diff
@@ -130,16 +132,21 @@ class Ads {
 
         billTheLizard.projectsHandler.enable('cheshirecat');
         billTheLizard.executor.register('catlapseIncontentBoxad', () => {
-          const slots = Object.keys(context.get('slots'))
-            .filter(slotName => slotName.indexOf('incontent_boxad_') === 0);
+          const slot = `incontent_boxad_${Object.keys(this.cheshirecatPredictions).length + 2}`;
```
Comment (fraszczakszymon) [inline app/modules/ads/module.js]: Why +2?

Comment (dmnsobczak) [reply app/modules/ads/module.js]: +1: Next incontent_boxad
+2: IncontentBoxads are counted from 1, not 0

Comment (fraszczakszymon) [reply app/modules/ads/module.js]: 🤦‍♂️ fair enough

Comment (fraszczakszymon) [reply app/modules/ads/module.js]: Would you please extract `Object.keys(this.cheshirecatPredictions).length + 2` to separate method `getNextIncontentIndex`?

**Review by fraszczakszymon — APPROVED** [review]

File: app/modules/ads/module.js (hunk 1 above)
Comment (dmnsobczak) [reply app/modules/ads/module.js]: Good idea, done.

**Review by fraszczakszymon — APPROVED** [review]

**Pull request merged**